### PR TITLE
debian,cmake,cephsqlite: hide non-public symbols

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by batrick** [comment]
LGTM. You can rebase now.

**Comment by tchaikov** [comment]
@batrick actually, i am not quite satisfied with the outcome of this changeset. because there still a bunch of weak symbols exposed by libcephsqlite even with this change. most of them are from boost, but i cannot find a simple way to hide them. `dpkg-gensymbols` finds them when looking at libcephsqlite.so, and considers them new symbols introduced by the latest version of `libcephsqlite.so` on top of the ones declared in the .symbols file. this does not hurt, as no symbols are *removed*. but still, it's not perfect. i also looked into https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html, but it seems it does not give us more control over the symbol visibility in addition to what i am using in this changeset. it's just more convenient.

**Comment by tchaikov** [comment]
jenkins test make check

**Review by batrick — CHANGES_REQUESTED** [review]
> @batrick actually, i am not quite satisfied with the outcome of this changeset. because there still a bunch of weak symbols exposed by libcephsqlite even with this change. most of them are from boost, but i cannot find a simple way to hide them. 

Is this is a consequence of the shared library not being versioned? I would think other libraries like librados would have the same issue?

File: debian/control [hunk 1]
```diff
@@ -227,7 +227,6 @@ Package: ceph-mgr
 Architecture: linux-any
 Depends: ceph-base (= ${binary:Version}),
          ceph-mgr-modules-core (= ${binary:Version}),
-         libsqlite3-mod-ceph,
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply debian/control:230]: thanks! will add this tag in the commit message.

**Comment by tchaikov** [comment]
> > @batrick actually, i am not quite satisfied with the outcome of this changeset. because there still a bunch of weak symbols exposed by libcephsqlite even with this change. most of them are from boost, but i cannot find a simple way to hide them.
> 
> Is this is a consequence of the shared library not being versioned? 

no, they are orthogonal. 

> I would think other libraries like librados would have the same issue?

yes. we never added the .symbols file for librados or other packaged public shared libraries, not because `dh_makeshlibs` is able to figure out the shlib deps correctly for us. just because we always hardwire the dependencies in debian/control. `libcephsqlite` is the first one. because `dh_makeshlibs` failed to do set the subvar for us, and more importantly, to ~~replace the light bulb~~ root cause of failure caused by the buggy packaging took the efforts of 3 engineers. so i felt obliged to figure it out.

**Comment by tchaikov** [comment]
@batrick added the "Fixes" tag in the commit message and repushed.

**Review by batrick — APPROVED** [review]

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-03-25_06:40:11-rados-wip-kefu-testing-2021-03-25-1305-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-03-25_14:56:28-rados-wip-kefu-testing-2021-03-25-2042-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-03-26_05:32:58-rados-wip-kefu-testing-2021-03-26-1134-distro-basic-smithi/

**Pull request merged**